src/controller/java/src/chip/devicecontroller/model/ClusterState.java

```diff
@@ -51,6 +57,41 @@ public Optional<Long> getDataVersion() {
     return dataVersion;
   }
 
+  /**
+   * Convenience utility for getting all attributes in Json string format.
+   *
+   * @return all attributes in Json string format., or empty string if not found.
+   */
+  public String getAttributesJson() {
+    JSONObject combinedObject = new JSONObject();
+    ArrayList<JSONObject> attributeJsons =
+        attributes
+            .values()
+            .stream()
+            .map(it -> it.getJson())
+            .filter(it -> it != null)
+            .collect((Collectors.toCollection(ArrayList::new)));
+    for (JSONObject attribute : attributeJsons) {
+      Iterator<String> result = attribute.keys();
+      attribute
+          .keys()
+          .forEachRemaining(
```

Comment:
do we need forEachRemaining? does `for (key in attribuyte.keys()` not work? This feels off ... was hoping kotlin code could be made clearer.

Reply:
for (key in attribute.keys() not working...